running-code/execute-package.md

```diff
@@ -0,0 +1,209 @@
+```
+
+(executable-packages)=
+## 2. Executable packages
+
+The `-m` flag as described above only works for Python modules (files), but does not work for Python (sub-)packages (directories).
+This means that you cannot execute a command using only the name of your package when it is structured to use directories
+
+Once your package grows, the top-level name `my_program` turns into a directory.
+(See [Python Package Structure](https://www.pyopensci.org/python-package-guide/package-structure-code/python-package-structure.html)
+for when and how to create a package structure).
+
+```
+project/
+└── src/
+    └── my_program/
+        ├── __init__.py
+        └── greeting/
+            ├── __init__.py
+            └── hello.py
+```
+
+However, the directory is not executable.
+
+```bash
+python -m my_program
+python: No module named my_program.__main__; 'my_program' is a package and cannot be directly executed
+```
+
+Initially, Python seems to tell you that the directory names, including your top-level package name,
+cannot be directly executed. But the error message contains the hint that you need to make this run properly.
+
+[Earlier you learned](execute-script-name-eq-main) that `if __name__ == "__main__":` can protect parts of your
+Python file from executing when it is imported, making that conditional change the file's behavior when used as
+a script vs when used as a module. There is a very similar concept that can be applied to Python directories.
+
+You may already know that a directory that contains an [`__init__.py` module](https://www.pyopensci.org/python-package-guide/tutorials/installable-code.html#what-is-an-init-py-file)
+becomes a valid `import` target and that whenever the directory is imported, the code in the `__init__.py` is executed.
+There is another special file Python directories can contain: [`__main__.py` module](https://docs.python.org/3/library/__main__.html#module-__main__).
+Any package that contains a `__main__.py` can be execued with `python -m` exactly like a Python module. When a
+Python package (directory) is executed, the code in `__main__.py` is executed, as if it was the target of the `-m`.
+
+In this way a Python directory can segment its import behaviour from its command behaviour by using both
+`__init__.py` and `__main__.py` in a very similar way to how a Python file segments this behaviour using
+`if __name__ == "__main__":`.
+
+If we add to the earlier package structure, we can make the original execution command work.
+
+```
+project/
+└── src/
+    └── my_program/      <-- directories with init and main can be imported or executed
+        ├── __init__.py
+        ├── __main__.py
+        └── greeting/    <-- directories with init an no main can be imported but not executed
+            ├── __init__.py
+            └── hello.py
+```
+
+```python
+# project/src/my_program/__main__.py
+from .greeting.hello import shiny_hello
+
+# notice top-level evaluations are not guarded by any conditional
```

Comment:
This comment may be a bit confusing for a learner. I'm wondering if it would be better to remove.